djautotask/sync_rest.py

```diff
@@ -206,8 +209,13 @@ def _set_datetime_attribute(self, instance, attribute_name):
 
     def fetch_sync_by_id(self, instance_id):
         api_instance = self.get_single(instance_id)
-        instance, created = \
-            self.update_or_create_instance(api_instance['item'])
+        try:
+            instance, created = \
+                self.update_or_create_instance(api_instance['item'])
+        except TypeError as e:
+            logger.warning('{}'.format(e))
+            instance = None
```

Comment:
Instead of trying to create something we know will fail (because `get_single` returned a null instance) lets check if it exists and raise a `AutotaskRecordNotFoundError` if it doesn't. That way calling code can pick that up and handle it as it chooses. Otherwise we have to adapt all current code to detect when `fetch_sync_by_id` returns `None`, which is already a big change as no code expects that to ever return `None`. 

Then just make sure everything that calls `fetch_sync_by_id` catches `AutotaskRecordNotFoundError`. I'm pretty sure its just callbacks and maybe one or two other things. Not a lot.

To be fair this is Autotasks fault. If we request a record and it doesn't exist, it should be kind enough to give us a 404. Yeesh.

Reply:
Regarding `api_instance`, it should return normally as follows (from the responses of postman):

```
{
    "items": [],
    "pageDetails": {
        ...
    }
}
```

In other words, in usual cases, api_instance['item'] should be empty list, and it can be iterable. 

But, to be surprise, it seems return as follows occasionally:

```
{
    "items": null,
    "pageDetails": {
        ...
    }
}
```

Furthermore, regarding return of `fetch_sync_by_id`:
In the case that get_single returns instance, the code return instance. So, if it doesn't exist, it can return None logically.
Luckily, their return is only used in one line (line 620) for now.

![image](https://user-images.githubusercontent.com/73486314/134754958-bd8ed6b0-ef8b-411a-9fb9-3bd37bd84644.png)

So, I hope to hear your opinion once again. 

However, I have got some idea from your comments, so I fixed the code a little bit more and pushed that, too.

Reply:
Ok, this is worth talking about. Can you provide more information? Under what conditions does this occational case happen? Can you reproduce it reliably? These are important things to know before deciding what to do.

Reply:
Does it only return null if we use `get_single/fetch_sync_by_id`? If so then my previous point about `AutotaskRecordNotFoundError` still stands.